src/main/resources/jupyter/safe-mode.js

```diff
@@ -0,0 +1,105 @@
+// Adapted from the All of Us Researcher Workbench "Playground Mode"
+// https://github.com/all-of-us/workbench/blob/master/api/cluster-resources/playground-extension.js
+
+// TODO: the following is required until IA-979 is implemented
+// To deploy on a Leonardo cluster:
+// 1. copy to a public GCS location
+// 2. issue a PUT request to <leonardo>/api/cluster/<billing project>/<cluster name>
+//  with a name:path key value pair set in userJupyterExtensionConfig.nbExtensions
+
+// In "Safe Mode", changes are not saved back to GCS. This extension makes
+// minor UI tweaks to differentiate this mode from normal Jupyter usage, and
+// also removes/hides controls relating to persistence. Technically
+// this does not stop autosave from continuing to happen in the background, but
+// the intended use of this plugin is in a separate space from normal operation
+// which does not support localization features.
+
+define([
+    'base/js/namespace'
+], (Jupyter) => {
+  const load = () => {
+    // TODO: query welder for when to enable (IA-979).   always-on, for now
+    const enabled = true;
+    if (!enabled) {
+      return;
+    }
+
+
+    //
+    // Disable UI controls
+    //
+
+
+    // Save Notebook button
+    //"notbook" is an intentional typo to match the Jupyter UI HTML.
+    $('#save-notbook').remove();
+
+    // New Notebook menu tree
+    $('#new_notebook').remove();
+
+    // Open... menu item
+    $('#open_notebook').remove();
+
+    // Make a Copy... menu item
+    $('#copy_notebook').remove();
+
+    // Save as... menu item
+    $('#save_notebook_as').remove();
+
+    // Save and Checkpoint menu item
+    $('#save_checkpoint').remove();
+
+    // Revert to Checkpoint menu tree
+    $('#restore_checkpoint').remove();
+
+    // A little cleanup: remove the first two dividers in the file menu
+    // which are no longer dividing anything
+    $('#file_menu')
+        .append(
+            '<style>' +
+              // TODO this does not remove the first divider.
+              // instead it removes the whole menu.  Why?
+              //'li:nth-of-type(1) { display: none; } ' +
```

Comment:
I am still a long way from understanding CSS

Reply:
It might be easier to use a jquery selector to capture this.. I'm launching a cluster now to play around with it. The other style element approach I used because I believe those notification divs aren't rendered until they are active.

Reply:
Try this out:

```
$("#file_menu .divider")[2].remove()
```

This is a jQuery selector if you want to look it up. jQuery is definitely out of vogue so don't worry about learning too much about it - but you're likely to encounter it on older projects like this.

Reply:
(but the internal piece is a valid CSS selector in this case)

Reply:
Thanks, that works.